test failed: Base.samefile(Libdl.dlpath(dlls[1]),dlls[1])

```julia
[workerm@evo git_julia_new]$ export JULIA_CPU_CORES=2
[workerm@evo git_julia_new]$ make testall
cp /0_JULIA_COMPILED_0/git_julia_new/usr/lib/julia/sys.ji local.ji && /0_JULIA_COMPILED_0/git_julia_new/usr/bin/julia -J local.ji -e 'true' && rm local.ji
    JULIA test/all
	From worker 3:	     * linalg2              in  44.13 seconds
	From worker 3:	     * linalg3              in  13.87 seconds
	From worker 3:	     * linalg4              in   5.30 seconds
	From worker 3:	     * linalg/lapack        in   1.93 seconds
	From worker 2:	     * linalg1              in  86.11 seconds
	From worker 2:	     * linalg/tridiag       in   5.65 seconds
	From worker 2:	     * linalg/bidiag        in   8.87 seconds
	From worker 2:	     * linalg/diagonal      in   7.55 seconds
	From worker 2:	     * linalg/pinv          in   7.22 seconds
	From worker 2:	     * linalg/givens        in   2.19 seconds
	From worker 2:	     * linalg/cholesky      in   9.23 seconds
	From worker 2:	     * linalg/lu            in  13.63 seconds
	From worker 2:	     * linalg/symmetric     in   6.39 seconds
	From worker 3:	     * linalg/triangular    in 101.71 seconds
	From worker 2:	     * linalg/arnoldi       in  21.16 seconds
	From worker 2:	     * keywordargs          in   2.43 seconds
	From worker 3:	     * core                 in  26.04 seconds
	From worker 3:	     * strings              in  31.62 seconds
	From worker 2:	     * numbers              in  66.90 seconds
	From worker 2:	     * dict                 in  12.57 seconds
	From worker 2:	     * hashing              in   5.86 seconds
	From worker 2:	     * remote               in   0.42 seconds
	From worker 2:	     * iobuffer             in   1.01 seconds
	From worker 2:	     * staged               in   0.65 seconds
	From worker 3:	     * dates                in  54.44 seconds
	From worker 3:	     * tuple                in   1.00 seconds
	From worker 2:	     * arrayops             in  33.67 seconds
	From worker 2:	     * reduce               in   4.77 seconds
	From worker 2:	     * reducedim            in  17.46 seconds
	From worker 2:	     * random               in  13.28 seconds
	From worker 2:	     * intfuncs             in   0.87 seconds
	From worker 2:	     * simdloop             in   1.21 seconds
	From worker 2:	     * blas                 in   1.57 seconds
	From worker 2:	     * sparse               in  66.79 seconds
	From worker 3:	     * subarray             in 165.63 seconds
	From worker 3:	     * copy                 in   2.12 seconds
	From worker 3:	     * math                 in  12.49 seconds
	From worker 2:	     * bitarray             in  64.55 seconds
	From worker 2:	     * functional           in   1.37 seconds
	From worker 2:	     * operators            in   0.20 seconds
	From worker 3:	     * fastmath             in   3.96 seconds
	From worker 2:	     * path                 in   1.03 seconds
	From worker 3:	     * ccall                in   2.29 seconds
	From worker 2:	     * unicode              in   4.13 seconds
	From worker 3:	     * bigint               in   3.51 seconds
	From worker 3:	     * statistics           in  12.40 seconds
	From worker 3:	     * spawn                      [stdio passthrough ok]
	From worker 3:	 in   8.00 seconds
exception on 3: ERROR: LoadError: test failed: Base.samefile(Libdl.dlpath(dlls[1]),dlls[1])
 in expression: Base.samefile(Libdl.dlpath(dlls[1]),dlls[1])
 in error at error.jl:21
 in default_handler at test.jl:29
 in do_test at test.jl:52
 in include at ./boot.jl:252
 in runtests at /0_JULIA_COMPILED_0/git_julia_new/test/testdefs.jl:79
 in anonymous at multi.jl:838
 in run_work_thunk at multi.jl:589
 in anonymous at task.jl:838
while loading /0_JULIA_COMPILED_0/git_julia_new/test/backtrace.jl, in expression starting on line 22
	From worker 2:	     * sorting              in  32.81 seconds
ERROR: LoadError: LoadError: test failed: Base.samefile(Libdl.dlpath(dlls[1]),dlls[1])
 in expression: Base.samefile(Libdl.dlpath(dlls[1]),dlls[1])
 in anonymous at task.jl:1394
while loading /0_JULIA_COMPILED_0/git_julia_new/test/backtrace.jl, in expression starting on line 22
while loading /0_JULIA_COMPILED_0/git_julia_new/test/runtests.jl, in expression starting on line 5

	From worker 3:	     * backtrace           Makefile:9: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:477: recipe for target 'testall' failed
make: *** [testall] Error 2
[workerm@evo git_julia_new]$ 
```

Arch Linux:

```julia
julia> versioninfo()
Julia Version 0.4.0-dev+4829
Commit 9a47c3b (2015-05-14 14:22 UTC)
Platform Info:
  System: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Core(TM) i3-2310M CPU @ 2.10GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3

julia> 

```

--- a/test/backtrace.jl
+++ b/test/backtrace.jl
@@ -19,8 +19,11 @@ dlls = Libdl.dllist()
 @test !isempty(dlls)
 @test length(dlls) > 3 # at a bare minimum, probably have some version of libstdc, libgcc, libjulia, ...
 if @unix? true : (Base.windows_version() >= Base.WINDOWS_VISTA_VER)
-    @test Base.samefile(Libdl.dlpath(dlls[1]), dlls[1])
-    @test Base.samefile(Libdl.dlpath(dlls[end]), dlls[end])
+    for dl in dlls
+        if isfile(dl)
+            @test Base.samefile(Libdl.dlpath(dl), dl)
+        end
+    end
 end
 @test length(filter(dlls) do dl
         return ismatch(Regex("^libjulia(?:.*)\.$(Libdl.dlext)(?:\..+)?\$"), basename(dl))

